src/components/molecules/TableHeader/TableHeader.tsx

```diff
@@ -138,60 +149,107 @@ const TableHeader: React.FC<TableHeaderProps> = ({
     };
   }, [leaveSeat]);
 
+  const updateTableTitle = useCallback(
+    async (newTableTitle) => {
+      if (!user) return;
+
+      const venueTables = tables.map((table) => {
+        if (table.reference === tableOfUser?.reference) {
+          return { ...table, title: newTableTitle };
+        }
+        return table;
+      });
+
+      // Ideally we want to do this only when config.tables doesn't exist. Otherwise we want to update only a single table.
+      await updateVenue_v2(
+        { name: venueName, tables: venueTables },
+        user
+      ).catch(() => {
```

Comment:
Can you please explain the logic/intent of this (including the note made in the comment). Not really following it at the moment. Why wouldn't we just be providing the table we want to update + the new title?

Would a more specific/focused backend function make more sense here than this 'update the entire venue' function?

Reply:
For me personally, it does make more sense to use another function. I think there should be a transaction that updates only a single table with `merge` option, if that's possible. Right now, I'm replacing the whole array but updating the title and capacity only of the current table in that array. This can be quite expensive and heavy if there are 100 tables, I think?

Reply:
Maybe I'm just being paranoid with all the performance issues we had.

Reply:
> Right now, I'm replacing the whole array but updating the title and capacity only of the current table in that array. This can be quite expensive and heavy if there are 100 tables, am I wrong?

Having to do this from the frontend is probably suboptimal due to extra bandwidth required to send the request/etc. Not sure it would be world breakingly bad, but still feels kind of hacky to me.

---

I'd personally implement the backend function where you just provide the minimal details required, then use a firebase transaction to read the venue, update the required part in the tables array, then write it back again.

I don't believe there are any magic firebase functions to allow us to merge into the array nicely 'automagically'. There is `arrayUnion` and similar.. but not sure how they will work with complex objects (eg. table config objects) rather than primitive types (eg. string)

Reply:
Makes sense. So let's do this as a backend function as outlined in https://github.com/sparkletown/sparkle/pull/1377#discussion_r632345020

Reply:
Sounds good. Makes more sense to be that way.

Reply:
Fixed in: https://github.com/sparkletown/sparkle/pull/1377/commits/c222264f9875e83b03a9a64b49c9ae1d57e9401c